src-self-hosted/Module.zig

```diff
@@ -2927,6 +2927,11 @@ pub fn resolvePeerTypes(self: *Module, scope: *Scope, instructions: []*Inst) !Ty
             continue;
         }
 
+        if ((prev_inst.ty.zigTypeTag() == .ComptimeInt and next_inst.ty.isInt()) or (next_inst.ty.zigTypeTag() == .ComptimeInt and prev_inst.ty.isInt())) {
+            prev_inst = next_inst;
+            continue;
+        }
```

Comment:
This is the wrong logic when next_inst is a comptime int. You should probably add test coverage for this